### ISSUE TITLE
Ask students to set personalizations when they sign in

### ISSUE THREAD

**Issue by benetechMatt** [benetech/MathShare#974]
Ask students to set personalizations when they sign in
Reminders of what changes they can do.  Do they want to change the current settings?  Needs can change day to day. Reminding students helps them
Add a “do not remind” check box. This can be undone via settings

**Comment by benetechMatt** [comment]
![mockup of personalization settings. Title of "Personalization Settings". The second line has the label "Speak text" and a toggle switch with On and Off as options. The next row has "font" with a drop down. Towards the bottom is the text 'These settings can be changed under personalization settings under your user profile icon'. below that is a button that says "save"](https://user-images.githubusercontent.com/49413678/75603654-0ee98280-5a96-11ea-9912-a18d87f38a7f.png)

**Comment by benetechMatt** [comment]
settings are saved into the user's dashboard and automatically adjust to meet the student's needs.

**Comment by alexrcabral** [comment]
I'd like some clarification on how settings "automatically adjust to meet the student's needs".
1) How are they determined to be a need
2) what adjustments are made, and 
3) do they change over time?

**Comment by johnhbenetech** [comment]
This mockup looks nothing like the other personalization settings tasks. Please consolidate to a final agreed upon design

**Comment by benetechMatt** [comment]
The thinking with this is to give students a subset of settings that they will most likely want to change or enable.
I'm assuming most students will want to keep TTS on, but not all.  By having this settings question during sign up the goal is to make students aware that they can turn it off and that there are other settings they can adjust.

@alexrcabral should this be done someplace else instead of the sign up process?  Should we instead focus on just making users aware of the personalization settings?

@johnhbenetech this is meant to be a smaller, separate screen students see during sign up instead of the potentially overwhelming list of settings page

**Comment by alexrcabral** [comment]
@benetechMatt This makes sense to do during the sign up process so students don't have to dig to find accessibility settings. I agree that it shouldn't be overwhelming, so not throwing the entire expanded personalization settings page at them from the get go, but it does need to be reworked. 

Other considerations: The toggle is incorrectly implemented. When clicked so the circle is to the right, that is the standard "active" mode for the toggle, and denotes an action (e.g. turning on TTS). The "Off" label should be on the left, and the "On" label should be on the right.

**Comment by benetechMatt** [comment]
@alexrcabral from a prior conversation, should this be changed to a checkbox since there will not be an immediate change to the page?

Should all messages/text have TTS or only focus on the math solution process?

**Comment by alexrcabral** [comment]
@benetechMatt 
Ideally it would be a toggle that does have immediate impact on the page. 

This should be based on more granular controls in the main settings — eg all messages, just steps, or just current step. Default should be on all steps, both math and descriptions. 

To clarify existing issues: 
1. “Do not remind” checkbox is missing from the mock up
2. Does this happen every single time they sign in? You’ve switched between sign in and sign up and I’m not sure which you’re going for here 
3. The toggle is still incorrectly implemented
4. Need clarification on how it “automatically adjusts to meet the student’s needs” 
    - How is it determined to be a need
    - What adjustments are made
    - Do they change over time
5. John’s point on a unified design — even if it’s not the same modal being used, it needs to be consistent so it’s understandable and usable

**Comment by jscholes** [comment]
#### TTS Option

> I'm assuming most students will want to keep TTS on

It's really important that the purpose of this function is surfaced for a screen reader user.  There cannot be a case where a user thinks that they need to leave TTS enabled even though they have a screen reader, and then ends up not being able to hear the screen reader because the application is talking.

#### Cancel Mechanism

There doesn't seem to be a "Cancel" mechanism.  We should discuss what that will look like and how it should behave:

* If a user has no additional needs, do they need to explicitly turn everything off and then hit "Save"?
* How do we decide which options are enabled by default?
* If options are enabled by default, should cancelling imply that the user doesn't want them?

**Comment by alexrcabral** [comment]
As an update from the call today, this is being moved to the story on TTS personalization settings #1004. (Not worth the added modal and potential barrier after signing up for minimal personalization settings)
